oceannavigator/frontend/src/components/TimePicker.jsx

```diff
@@ -117,142 +109,134 @@ class TimePicker extends React.Component {
-                    MaxMonth: new Date(
-                      map[max].getTime() +
-                      map[max].getTimezoneOffset() * 60 * 1000
-                    ),
-                    i18n: {
-                      year: _("Year"),
-                      prevYear: _("Previous Year"),
-                      nextYear: _("Next Year"),
-                      next12Years: _("Jump Forward 12 Years"),
-                      prev12Years: _("Jump Back 12 Years"),
-                      nextLabel: _("Next"),
-                      prevLabel: _("Prev"),
-                      buttonText: _("Open Month Chooser"),
-                      jumpYears: _("Jump Years"),
-                      backTo: _("Back to"),
-                      months: [
-                        _("Jan."),
-                        _("Feb."),
-                        _("Mar."),
-                        _("Apr."),
-                        _("May"),
-                        _("June"),
-                        _("July"),
-                        _("Aug."),
-                        _("Sep."),
-                        _("Oct."),
-                        _("Nov."),
-                        _("Dec.")
-                      ]
-                    }
-                  });
-                  break;
-                case "hour":
-                  $(this.refs.picker).datepicker({
-                    Button: false,
-                    dateFormat: "dd MM yy",
-                    onClose: this.pickerChange,
-                    minDate: new Date(map[min]),
-                    maxDate: new Date(map[max])
-                  });
-                  $(this.refs.picker).datepicker(
-                    "option",
-                    "minDate",
-                    new Date(map[min])
-                  );
-                  $(this.refs.picker).datepicker(
-                    "option",
-                    "maxDate",
-                    new Date(map[max])
-                  );
-                  break;
-                case "day":
-                default:
-                  $(this.refs.picker).datepicker({
-                    Button: false,
-                    dateFormat: "dd MM yy",
-                    onClose: this.pickerChange,
-                    minDate: new Date(map[min]),
-                    maxDate: new Date(map[max])
-                  });
-
-                  $(this.refs.picker).datepicker(
-                    "option",
-                    "minDate",
-                    new Date(map[min])
-                  );
-                  $(this.refs.picker).datepicker(
-                    "option",
-                    "maxDate",
-                    new Date(map[max])
-                  );
-                  break;
-              }
-              this.pickerChange();
-            }.bind(this)
+    // eslint-disable-next-line max-len
+    GetTimestampsPromise(props.dataset, props.variable).then(timestampResult => {
+      const data = timestampResult.data;
+
+      let map = {};
```

Comment:
all below this was copy-pasted from the original implementation, I just wrapped it all in the cache-able timestamps promise.